src/wp-includes/option.php

```diff
@@ -196,6 +169,33 @@ function get_option( $option, $default_value = false ) {
 			$value = wp_cache_get( $option, 'options' );
 
 			if ( false === $value ) {
+				// Prevent non-existent options from triggering multiple queries.
+				$notoptions = wp_cache_get( 'notoptions', 'options' );
+
+				// Prevent non-existent `notoptions` key from triggering multiple key lookups.
+				if ( ! is_array( $notoptions ) ) {
+					$notoptions = array();
+					wp_cache_set( 'notoptions', $notoptions, 'options' );
+				}
+
+				if ( isset( $notoptions[ $option ] ) ) {
```

Comment:
Based on the previous `if`, if `$notoptions` isn't an array, then we already know this clause will be `false`. So a bit more efficient to make it an `elseif` instead of checking.
```suggestion
				} elseif ( isset( $notoptions[ $option ] ) ) {
```